app/routes/circulars/circulars.server.ts

```diff
@@ -110,6 +110,62 @@ export async function search({
 
   const [startTime, endTime] = getValidDates(startDate, endDate)
 
+  const esQuery = query
+    ? {
+        function_score: {
+          query: {
+            bool: {
+              must: [
+                {
+                  query_string: {
+                    query: `*${query}*`,
+                    fields: [
+                      'eventId^4',
+                      'synonyms^3',
+                      'subject^2',
+                      'body^1',
+                      'submitter',
+                    ],
+                    fuzzy_max_expansions: 50,
+                  },
+                },
+                {
+                  range: {
+                    createdOn: {
+                      gte: startTime,
+                      lte: endTime,
+                    },
+                  },
+                },
+              ],
+            },
+          },
+          functions: [
+            {
+              filter: { exists: { field: 'eventId' } },
+              weight: 4,
+            },
+            {
+              filter: { exists: { field: 'synonyms' } },
+              weight: 3,
+            },
+            {
+              filter: { exists: { field: 'subject' } },
+              weight: 2,
+            },
+            {
+              filter: { exists: { field: 'body' } },
+              weight: 1,
+            },
+          ],
+          score_mode: 'sum',
+          boost_mode: 'sum',
```

Comment:
Please explain what this does. I can't find it in the documentation.

Reply:
If you look at the next level up, you will see that it is part of the function_score. function_score can take an array called functions. functions are applied to each result. It always has two properties in this case, filter and weight. Filter sets a filter that determines whether or not the function should be applied and weight assigns a weight to the function. This ranks the importance of the function being applied. The higher the weight, the more importance. These are then used to calculate the final relevance score. If it has a higher weight, that contributes more to the score.

score_mode specifies how the score should be calculated to come up with the final relevance score. So all the scores from all the functions are added to make the final_score. 

boost_mode specifies how the scores should be combined. Since I am using sum, the individual function scores are summed. 
    
https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-function-score-query.html
https://opster.com/guides/elasticsearch/search-apis/elasticsearch-function-score/

Reply:
I think that what you intend here is to apply a given weight depending on whether the search term matches the body, or the subject, or the event ID, etc. But is that what this actually does? As I read it, I would guess that it adds the weight if the _matching document_ contains a body, or a subject, or an event ID, etc. --- and most documents will have all of those fields.